src/components/Table/useColumns.ts

```diff
@@ -1,11 +1,41 @@
-import { Dispatch, SetStateAction, useState } from "react";
+import { camelCase } from "change-case";
+import { Dispatch, SetStateAction, useEffect, useState } from "react";
+import { useSessionStorage } from "src/hooks/useSessionStorage";
 import { GridColumn, Kinded } from "./GridTable";
 
 export function useColumns<R extends Kinded, S = {}>(
   tableColumns: GridColumn<R, S>[],
+  maybeStorageKey?: string,
 ): [GridColumn<R, S>[], Dispatch<SetStateAction<GridColumn<R, S>[]>>] {
-  const [columns, setColumns] = useState<GridColumn<R, S>[]>(
-    tableColumns.filter((column) => (column.canHide ? column.visible : true)),
+  const { selectedColumns, hideColumns } = tableColumns.reduce(
+    (acc, column) => {
+      // Only include options that can be hidden and have the `name` property defined.
+      if (!column.name || column.name.length === 0) {
+        console.warn("Column is missing 'name' property required by the Edit Columns button", column);
+        return acc;
+      }
+      // Add hide-able columns
+      if (column.canHide) {
+        acc.hideColumns.push(column.name);
+      }
+      // Add selected columns
+      if (!column.canHide || (column.canHide && column.visible)) {
```

Comment:
@pgamboa07 / @bdow the `column.visible` flag is a little surprising to me; does the user toggle that on/off to get the column shown/not shown in response to `setColumns` called? like:

* User clicks show/hide column
* `EditColumnsButton` calls `setColumns`
* The page re-memozies its `createColumns` b/c each column has a `GridColumn.visible = columns.includes(me)`?

Is that how it currently works?

It seems like it'd be nice to avoid breaking the `createColumns` `useMemo` every time a user shows/hides a column, but I guess realistically as our table is currently rewritten we need to rerender every row anyway...

If the `EditColumnsButton` is now taking a `selectedColumns` prop, can the `column.visible` go away?

Reply:
The visible flag is used to identify which hide-able columns are visible/selected by default on initial setup